store/cachekv/store.go

```diff
@@ -104,43 +130,34 @@ func (store *Store) Write() {
 	}
 
 	// We need a copy of all of the keys.
-	// Not the best, but probably not a bottleneck depending.
-	keys := make([]string, 0, len(store.cache))
+	// Not the best. To reduce RAM pressure, we copy the values as well
+	// and clear out the old caches right after the copy.
+	sortedCache := make([]cEntry, 0, len(store.cache))
 
 	for key, dbValue := range store.cache {
 		if dbValue.dirty {
-			keys = append(keys, key)
+			sortedCache = append(sortedCache, cEntry{key, dbValue})
 		}
 	}
-
-	sort.Strings(keys)
+	store.deleteCaches()
```

Comment:
I think after the `Write` call, the whole cache store is discarded usually?

Reply:
Oh interesting. If so:
- This is still a win, for getting the data discarded earlier
- We should just always clear everything, rather than the old capacity preservation code then